bcipy/display/paradigm/vep/vep_stim.py

```diff
@@ -0,0 +1,93 @@
+"""Display components for VEP"""
+from typing import List, Tuple
+
+from psychopy import visual  # type: ignore
+from psychopy.visual.shape import ShapeStim  # type: ignore
+
+from bcipy.display.components.layout import envelope
+from bcipy.display.paradigm.vep.layout import checkerboard
+
+
+class VEPStim:
+    """Represents a checkerboard of squares that can be flashed at a given
+    rate. Flashing is accomplished by inverting the colors of each square.
+
+    Parameters
+    ----------
+        layout - used to build the stimulus
+        code - A list of integers representing the VEP code for each box
+        colors - tuple of colors for the checkerboard pattern
+        center - center position of the checkerboard
+        size - size of the checkerboard, in layout units
+        num_squares - number of squares in the checkerboard
+    """
+
+    def __init__(self,
+                 win: visual.Window,
+                 code: List[int],
+                 colors: Tuple[str, str],
+                 center: Tuple[float, float],
+                 size: Tuple[float, float],
+                 num_squares: int = 4):
+        self.window = win
+        self.code = code
+
+        squares = checkerboard(squares=num_squares,
+                               colors=colors,
+                               center=center,
+                               board_size=size)
+        board_boundary = envelope(pos=center, size=size)
+
+        frame1_holes = []
+        frame2_holes = []
+        for square in squares:
+            square_boundary = envelope(pos=square.pos, size=square.size)
+            # squares define the holes in the polygon
+            if square.color == colors[0]:
+                frame2_holes.append(square_boundary)
+            elif square.color == colors[1]:
+                frame1_holes.append(square_boundary)
+
+        # Checkerboard is represented as a polygon with holes, backed by a
+        # simple square with the alternating color.
+        # This technique renders more efficiently and scales better than using
```

Comment:
That's good to know. The only other VEP I could find used the GradientStim. Is the performance gain due to the decrease in objects drawn overall?

Reply:
Yes, I tried various stim for this and there wasn't a significant difference. The main factor was the number of stim  drawn.